src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -798,17 +794,19 @@ public void scrollToPosition(String selector) {
     jsActions.scrollToSpecificElement(driver.findElement(By.cssSelector(selector)));
     PageObjectLogging.log("scrollToSelector", "Scroll to the web selector " + selector, true);
   }
-  // This scroll has been created because ad is not displayed if we scroll quickly to the Footer ADEN-4359
-  public void scrollToBottomLeaderboard(){
-    scrollToFooter();
-    wait.forElementVisible(By.cssSelector(".editarea"));
-    scrollToFooter();
-  }
 
-  public void clickOnAdImage(String slotName){
-    adSelector(slotName);
-    adSelector(slotName).click();
-    PageObjectLogging.log("clickOnAdImage", slotName + " is clicked", true);
+  // This scroll has been created because ad is not displayed if we scroll quickly to the Footer ADEN-4359
+  public void scrollToSlot(String slotName) {
+    switch(slotName) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1301)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)